### ISSUE TITLE
FatalErrorExceptions when using feature/relay branch with Lumen / Laravel 5.3 & 5.4

### ISSUE THREAD

**Issue by 4levels** [Folkloreatelier/laravel-graphql#82]
Hi,

I'm getting the following error when using Lumen / Laravel with this repository's feature/relay branch.

If I use Lumen / Laravel 5.3, the error is as follows (indicating that the router cannot be found as Lumen works differently):

    FatalErrorException` in ServiceProvider.php line 79:  
    Call to undefined method Laravel\Lumen\Application::pattern()

If I use Lumen / Laravel 5.4 (from my cloned repo with updated dependencies), I get

    ErrorException in ServiceProvider.php line 162:  
    Missing argument 2 for Folklore\GraphQL\ServiceProvider::Folklore\GraphQL\{closure}(),  
    called in /../vendor/illuminate/container/Container.php on line 678

This is not urgent as the ~1.0 branch works fine, even with Lumen / Laravel 5.4

Kind regards and keep up the great work here!

**Comment by dmongeau** [comment]
This should be fixed! Sorry about that

**Comment by 4levels** [comment]
Hi @dmongeau,

I just updated my fork with your latest commit to test.
So far there still seems to be an issue with Laravel's Passport library as this one seems to depend on Facades as well :-(

I've tried both Laravel / Lumen 5.3 and 5.4 but this doesn't make a difference.

```
RuntimeException in Facade.php line 234:
A facade root has not been set.
in Facade.php line 234
at Facade::__callStatic('extend', array('passport', object(Closure))) in PassportServiceProvider.php line 233
at Auth::extend('passport', object(Closure)) in PassportServiceProvider.php line 233
at PassportServiceProvider->registerGuard() in PassportServiceProvider.php line 83
at PassportServiceProvider->register() in Application.php line 174
at Application->register('Laravel\Passport\PassportServiceProvider') in app.php line 109
at require('/.../bootstrap/app.php') in index.php line 14
```

**Comment by dmongeau** [comment]
Could this be related to this line?

https://github.com/Folkloreatelier/laravel-graphql/blob/feature/relay/src/Folklore/GraphQL/GraphQLController.php#L34

What happen if you comment it and return null?

**Comment by 4levels** [comment]
Hi @dmongeau, I got in touch with the Laravel folks on IRC and found out that they had some deprecated calls to Facades in their ServiceProvider.
I create a PR that fixes this issue here: https://github.com/laravel/passport/pull/258
This caused a bug in this repo's routes.php to show up, so I created a PR as well: https://github.com/Folkloreatelier/laravel-graphql/pull/83

I can confirm that over here, with Lumen 5.4 and WITHOUT Facades, everything works as expected now!

Kind regards,

**Comment by dmongeau** [comment]
Merged #83 closing this issue
